### DISCUSSION TITLE
How do I debounce (warning) validation on type? (zod refine)

### DISCUSSION THREAD

**Discussion by filipjnc** [pablo-abc/felte#194] [Q&A]
I created a warning validation via zod.refine() for a text field that checks if the item exists already in the database:

```ts
const warnSchema = z.object({
    name: z.string().refine(
      async (name) => {
        return !(await checkNameExists({ name }));
      },
      { message: "Name exists already" }
    ),
  });

```

I'd like to debounce the warning validation so that it doesn't spam the server on type. The following setting does not work (probably only works with a custom validator:
`
debounced: {
    timeout: 1000,
},
`

**Comment by nicojones** [comment]
Would love to hear an answer by somebody here, @filipjnc seems such a basic use case that i'm surprised there aren't more threads asking the same.

We have two options at the moment:
- Block typing new "name" characters until `checkNameExists` returns (really bad!)
- Do the validation outside of Zod (really complicated)

All that's needed is that this particular `refine` gets completed asynchronously, but in the meantime further typing is allowed.
